src/shared/vendor.py

```diff
@@ -0,0 +1,382 @@
+        )
+        return customer
+    except (
+        InvalidRequestError,
+        APIConnectionError,
+        APIError,
+        RateLimitError,
+        IdempotencyError,
+        CardError,
+        StripeErrorWithParamCode,
+    ) as e:
+        logger.error("modify customer token", error=e)
+        raise e
+
+
+@retry(
+    wait=wait_exponential(multiplier=1, min=1, max=8),
+    stop=stop_after_attempt(4),
+    reraise=True,
+)
+def create_stripe_customer(
+    source_token: str, email: str, userid: str, name: str, idempotency_key: str
+) -> Customer:
+    """
+    Create a new Stripe Customer.
+    :param source_token:
+    :param email:
+    :param userid:
+    :param name:
+    :param idempotency_key:
+    :return: Customer
+    """
+    try:
+        customer = Customer.create(
+            source=source_token,
+            email=email,
+            description=userid,
+            name=name,
+            metadata={"userid": userid},
+            idempotency_key=idempotency_key,
+        )
+        return customer
+    except (
+        InvalidRequestError,
+        APIConnectionError,
+        APIError,
+        RateLimitError,
+        IdempotencyError,
+        CardError,
+        StripeErrorWithParamCode,
+    ) as e:
+        logger.error("create customer error", error=e)
+        raise e
+
+
+@retry(
+    wait=wait_exponential(multiplier=1, min=1, max=8),
+    stop=stop_after_attempt(4),
+    reraise=True,
+)
+def delete_stripe_customer(customer_id: str) -> Dict[str, Any]:
+    """
+    Delete a Stripe customer
+    :param customer_id:
+    :return: object
+    """
+    try:
+        deleted_customer = Customer.delete(sid=customer_id)
+        return deleted_customer
+    except (
+        InvalidRequestError,
+        APIConnectionError,
+        APIError,
+        RateLimitError,
+        StripeErrorWithParamCode,
+    ) as e:
+        logger.error("delete customer error", error=e)
+        raise e
+
+
+@retry(
+    wait=wait_exponential(multiplier=1, min=1, max=8),
+    stop=stop_after_attempt(4),
+    reraise=True,
+)
+def retrieve_stripe_customer(customer_id: str) -> Optional[Customer]:
+    """
+    Retrieve Stripe Customer
+    :param customer_id:
+    :return: Customer
+    """
+    try:
+        customer = Customer.retrieve(id=customer_id)
+        return customer
+    except (
+        InvalidRequestError,
+        APIConnectionError,
+        APIError,
+        RateLimitError,
+        StripeErrorWithParamCode,
+    ) as e:
+        logger.error("retrieve stripe customer error", error=e)
+        raise e
+
+
+# end Customer calls
+
+
+# begin Subscription calls
+@retry(
+    wait=wait_exponential(multiplier=1, min=1, max=8),
+    stop=stop_after_attempt(4),
+    reraise=True,
+)
+def build_stripe_subscription(
+    customer_id: str, plan_id: str, idempotency_key: str
+) -> Subscription:
+    """
+    Create a new Stripe subscription for a given customer
```

Comment:
The only thing to note in documentation is that it appears to be inconsistent between these methods and others below.